### PR TITLE
Add experimental templates that use the Single-project MSIX Packaging feature

### PR DESCRIPTION
**Pull request by alwu-msft**
This change adds templates to the Project Reunion VSIX that allow projects using the experimental Single-project MSIX Packaging feature to be created. It also adds an extension pack so that the corresponding VS extension that allows for such projects to be deployed/debugged (F5) from Visual Studio is automatically installed with the Project Reunion VSIX. Because this is an experimental feature, these additions are included only in preview builds of the Project Reunion VSIX.

### CODE REVIEW

File: dev/VSIX/Extension/ProjectReunion.Extension.csproj [hunk 1]
```diff
@@ -48,6 +48,9 @@
     <Content Include="LICENSE">
       <IncludeInVSIX>true</IncludeInVSIX>
     </Content>
+    <Content Condition="'$(EnableExperimentalVSIXFeatures)'=='true'" Include="ExtensionPack.vsext">
```
Comment (Scottj1s) [inline dev/VSIX/Extension/ProjectReunion.Extension.csproj:51]: what a great way to implement this and other future optional bits (might carry this over to WTS)

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/SingleProjectPackagedApp/MainWindow.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿#pragma once
+
+#pragma push_macro("GetCurrentTime")
+#undef GetCurrentTime
```
Comment (Scottj1s) [inline dev/VSIX/ProjectTemplates/Desktop/CppWinRT/SingleProjectPackagedApp/MainWindow.h:4]: should be redundant with pch #undef

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/SingleProjectPackagedApp/PropertySheet.props [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Scottj1s) [inline dev/VSIX/ProjectTemplates/Desktop/CppWinRT/SingleProjectPackagedApp/PropertySheet.props:1]: this is obsolete - can remove it (and vcxproj/filters pointers to it)

File: dev/VSIX/Extension/ExtensionPack.vsext [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "description": "Read about creating extension packs at https://aka.ms/vsextpack",
```
Comment (Scottj1s) [inline dev/VSIX/Extension/ExtensionPack.vsext:2]: should update this from the default, to something like "Project Reunion experimental extensions"

File: dev/VSIX/Extension/ExtensionPack.vsext [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  "version": "1.0.0.0",
+    "extensions": [
+        {
+            "vsixId": "Microsoft.SingleProjectMSIXPackagingTools",
```
Comment (Scottj1s) [inline dev/VSIX/Extension/ExtensionPack.vsext:6]: I don't see the connection between this vsixId and the two single-project templates - how does this work?

File: dev/VSIX/Extension/ProjectReunion.Extension.csproj [hunk 6]
```diff
@@ -220,6 +223,20 @@
       <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
       <IncludeOutputGroupsInVSIX>TemplateProjectOutputGroup%3b</IncludeOutputGroupsInVSIX>
     </ProjectReference>
+      <ProjectReference Include="..\ProjectTemplates\Desktop\CppWinRT\SingleProjectPackagedApp\WinUI.Desktop.CppWinRT.SingleProjectPackagedApp.csproj">
```
Comment (Scottj1s) [inline dev/VSIX/Extension/ProjectReunion.Extension.csproj:226]: indentation.  also, shouldn't these both have a  Condition="'$(EnableExperimentalVSIXFeatures)'=='true'" ?

Comment (alwu-msft) [reply dev/VSIX/Extension/ProjectReunion.Extension.csproj:226]: The condition is on the enclosing ItemGroup.

**Review by Scottj1s — APPROVED** [review]
looks good!

**Pull request merged**